rustls/src/crypto/mod.rs

```diff
@@ -505,6 +505,81 @@ pub trait ActiveKeyExchange: Send + Sync {
         Ok(complete_res)
     }
 
+    /// For hybrid key exchanges, returns a borrow of the paired classical key exchange.
+    ///
+    /// "Hybrid" means a key exchange algorithm which is constructed from two
+    /// (or more) underlying independent algorithms. Usually one is post-quantum-secure,
+    /// and the other is "classical".  See
+    /// <https://datatracker.ietf.org/doc/draft-ietf-tls-hybrid-design/10/>
+    ///
+    /// There is no requirement for a hybrid scheme (or any other!) to implement
+    /// `hybrid_pair`. It only enables an optimization.
+    ///
+    /// # Background
+    /// Rustls always sends a presumptive key share in its ClientHello, using
+    /// (absent any other information) the first item in [`CryptoProvider::kx_groups`].
+    /// If the server accepts the client's selection, it can complete the handshake
+    /// using that key share.  If not, the server sends a `HelloRetryRequest` instructing
+    /// the client to send a different key share instead.
+    ///
+    /// This costs an extra round trip, and wastes the key exchange computation
```

Comment:
Might be a smidge clearer to say:
```
// "This request costs an extra round trip ...
```
I think it better connects to the last sentence of the previous paragraph.